File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.go
func reverseWords(s string) string {
    var (
        ans string
        str string
    )
    
    for _, ch := range(s) {
        if(ch == ' ') {
            ans += str + " "
            str = ""
            continue
        }
        
        str = string(ch) + str
    }
    
    return ans + str
}

File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.cpp
class Solution {
public:
    string reverseWords(string s) {
        int start = 0;
        int end = s.size();

        string ans;
        string str;
        
        while(start < end) {
            if(s[start] == ' ') {
                reverse(str.begin(),str.end());
                ans += str;
                ans.push_back(' ');
                str.clear();
                start++;
                continue;
            }
            
            str.push_back(s[start++]);
        }
        
        reverse(str.begin(),str.end());
        ans += str;
        
        return ans;
    }
};